src/components/Highlighter.vue

```diff
@@ -1,13 +1,13 @@
 <template>
-  <span
-    :class="suit()"
-  >
+  <span :class="suit()">
     <component
       v-for="({ value, isHighlighted }, index) in parsedHighlights"
       :class="[isHighlighted && suit('highlighted')]"
       :key="index"
       :is="isHighlighted ? highlightedTagName : textNode"
-    >{{ value }}</component>
+    ><template v-if="value === ' '">
+      {{ value }}
+    </template><template v-else>{{ value }}</template></component>
```

Comment:
I'm having a hard time understanding this even after reading your summary. Can you help me a little bit? How is `v-if` different from `v-else`?

Reply:
It is different due to how white spaces are handled in the first block and the second block, 
![image](https://user-images.githubusercontent.com/2103975/89030721-5c159d00-d331-11ea-97dd-1df7203b97fe.png)

Reply:
Thanks @Kocal, indeed the difference is in the form. If text or moustache is directly without space to a tag, it will be rendered without wrapping spaces, however if it's on a newline, whitespace-only blocks will be preserved.

see https://github.com/vuejs/vue/issues/9208

Reply:
I've just changed it slightly  to make the difference more clear

leading & trailing whitespace | before this PR
---|---
<img width="1099" alt="Screenshot 2020-07-31 at 13 41 18" src="https://user-images.githubusercontent.com/6270048/89031859-9e3fde00-d333-11ea-940f-48b28757e9da.png"> | <img width="1100" alt="Screenshot 2020-07-31 at 13 41 01" src="https://user-images.githubusercontent.com/6270048/89031863-a13ace80-d333-11ea-8218-89638c2ffe3b.png">

Reply:
Thanks for the explanation @Kocal @Haroenv 
This is tricky 😅

Reply:
@Haroenv Since this fix isn't straightforward, can add a comment in the code that explains what you described in https://github.com/algolia/vue-instantsearch/pull/827#discussion_r463560724 including the Vue issue?

Reply:
Good call. Since you can't leave a comment in that place (you'd change the rendering), I found an alternative solution, where you can leave a comment